src/ids.rs

```diff
@@ -712,6 +712,100 @@ impl ChunkID {
     }
 }
 
+#[pyclass]
+#[derive(PartialEq, Eq, Clone, Hash)]
+pub(crate) struct SequesterServiceID(pub libparsec::types::SequesterServiceID);
+
+#[pymethods]
+impl SequesterServiceID {
+    #[new]
+    pub fn new(uuid: &PyAny) -> PyResult<Self> {
+        // Check if the PyAny as a hex parameter (meaning it's probably a uuid.UUID)
+        match uuid.getattr("hex") {
+            Ok(as_hex) => {
+                // Convert to string
+                let u = as_hex.extract::<&str>()?;
+                // Parse it as a Rust Uuid
+                match Uuid::parse_str(u) {
+                    Ok(as_uuid) => Ok(Self(libparsec::types::SequesterServiceID::from(as_uuid))),
+                    Err(_) => Err(PyValueError::new_err("Not a UUID")),
+                }
+            }
+            Err(_) => Err(PyValueError::new_err("Not a UUID")),
+        }
+    }
+
+    #[getter]
+    fn uuid<'p>(&self, py: Python<'p>) -> PyResult<&'p PyAny> {
+        let uuid = PyModule::import(py, "uuid")?;
+        let kwargs = vec![("hex", self.hex().unwrap())].into_py_dict(py);
+        match uuid.getattr("UUID")?.call((), Some(kwargs)) {
+            Ok(any) => Ok(any),
+            Err(err) => Err(PyValueError::new_err(err)),
+        }
+    }
+
+    #[classmethod]
+    fn from_bytes(_cls: &PyType, bytes: &[u8]) -> PyResult<Self> {
+        match uuid::Uuid::from_slice(bytes) {
+            Ok(uuid) => Ok(Self(libparsec::types::SequesterServiceID::from(uuid))),
+            Err(_) => Err(PyValueError::new_err("Invalid UUID")),
+        }
+    }
+
+    #[classmethod]
+    fn from_hex(_cls: &PyType, hex: &str) -> PyResult<Self> {
+        match hex.parse::<libparsec::types::SequesterServiceID>() {
+            Ok(entry_id) => Ok(Self(entry_id)),
+            Err(err) => Err(PyValueError::new_err(err)),
+        }
+    }
+
+    fn __richcmp__(&self, other: &Self, op: CompareOp) -> bool {
+        match op {
+            CompareOp::Eq => self.0 == other.0,
+            CompareOp::Ne => self.0 != other.0,
+            CompareOp::Lt => self.0 < other.0,
+            CompareOp::Gt => self.0 > other.0,
+            CompareOp::Le => self.0 <= other.0,
+            CompareOp::Ge => self.0 >= other.0,
+        }
+    }
+
+    fn __str__(&self) -> PyResult<String> {
+        Ok(self.0.to_string())
+    }
+
+    fn __repr__(&self) -> PyResult<String> {
+        Ok(format!("<SequesterServiceID {}>", self.0))
+    }
+
+    fn __hash__(&self, py: Python) -> PyResult<isize> {
+        hash_generic(&self.0.to_string(), py)
+    }
+
+    #[classmethod]
+    #[pyo3(name = "new")]
+    fn _class_new(_cls: &PyType) -> PyResult<Self> {
```

Comment:
```suggestion
    fn class_new(_cls: &PyType) -> PyResult<Self> {
```
We really need the `_` ?